Add access control to components implementing ListContext (#10188)
marmelab/react-admin: docs/useListController.md

    defaultTitle, // Translated title based on the resource, e.g. 'Posts'
    resource, // Resource name, deduced from the location. e.g. 'posts'
    refetch, // Callback for fetching the list data again
    // Authorization

>     canAccess, // Boolean indicating whether users have access to this resource

Reviewer: ```suggestion
    canAccess, // Boolean indicating whether the current user has access to this resource
```
```suggestion
    canAccess, // Boolean indicating whether users have access to this resource
```

    );
};
```

> ## Controlling Access To The List

Reviewer: ```suggestion
## Access Control
```

Let's keep the same title so that readers know where to find that information across hooks and components.

> Should your authProvider implements the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `useListController` hook will call it with the resource name to ensure users cannot access data they're no allowed to see. For instance, given the following code:

Reviewer: ```suggestion
Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `useListController` hook will call it with the resource name to determine if the current user can access data from the current resource.

 
  
    For instance, given the following code:
```
```suggestion
Should your authProvider implements the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `useListController` hook will call it with the resource name to ensure users cannot access data they're no allowed to see. For instance, given the following code:
```

> Should your authProvider implements the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `useListController` hook will call it with the resource name to ensure users cannot access data they're no allowed to see. For instance, given the following `<List>`:

Author: ```suggestion
Should your authProvider implements the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `useListController` hook will call it with the resource name to ensure users cannot access data they're no allowed to see. For instance, given the following code:
```

```tsx
import { useState } from 'react';
import { useListController } from 'react-admin';

const OfficeList = () => {
    const { canAccess, data, isPending } = useListController({ resource: 'offices' });

    if (isPending) return <div>Loading...</div>;
    if (!canAccess) return <div>You're not authorized to see the offices</div>;

    return (
      <ul>
          {data.map(record => (
              <li key={record.id}>{record.name}</li>
          ))}
      </ul>
    );
};
```

> e `authProvider.canAccess` method will be called times with the following parameters: `{ action: "read", resource: "offices" }`.

Reviewer: ```suggestion
`useListController` calls `authProvider.canAccess()` with the following parameters: 

- `{ action: "read", resource: "offices" }`.
```
